### ISSUE TITLE
[BUG] accessing arrays not working a expected 

### ISSUE THREAD

**Issue by Coupe70** [bitfocus/companion#2990]
### Is this a bug in companion itself or a module?

- [X] I believe this to be a bug in companion

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

`$(internal:custom_testarray)[x]` is not delivering the expected results.
It seems like it is acting like StringMid(testarray, x, 1), only delivering the one sign at index x
(including the separating commas) instead of the array element at index x.

Example array: 1,bla,test,4
`$(internal:custom_testarray)[0]` = 1  **---  ok**
`$(internal:custom_testarray)[1]` = , (comma) **--- not ok, should be 'bla'**
`$(internal:custom_testarray)[2]` = b **--- not ok, should be 'test'**
...

All given that I don't make mistakes in the way I setup the array.
"Arrays" in Companion seem to be just variables containing comma-separated values, is that correct?

### Steps To Reproduce

1. 'internal: Custom Variable: Set with expression' with expression `[1, "bla", "test", 4]` to set up an array 
2. 'internal: Custom Variable: Set with expression' to set a variable to `$(internal:custom_testarray)[x]`
3. display variable, for example as button text to check results

### Expected Behavior

Example array: 1,bla,test,4
`$(internal:custom_testarray)[0]` = 1  
`$(internal:custom_testarray)[1]` = bla
`$(internal:custom_testarray)[2]` = test
...

### Environment (please complete the following information)

```markdown
- OS: Windows 10
- Browser: Firefox
- Companion Version: 3.3.1-7001
```

**Comment by Julusian** [comment]
Yeah, all variables were being forced into strings when accessing them.
This is now fixed in the betas

**Comment by Coupe70** [comment]
Works for me in 3.4.0-7208 - thanks!!

**Comment by Coupe70** [comment]
@Julusian 
I'm not sure if this is 100% right in 3.4.0.
I do the following (as above):

1) 'internal: Custom Variable: Set with expression' with expression [1, "bla", "test", 4] to set up $(internal:custom_testarray)
2) 'internal: Custom Variable: Set with expression' to set a variable to $(internal:custom_testarray)[x]
3) display the variable, for example as button text to check results

So far so good, it works as expected.
When I look a the custom variables $(internal:custom_testarray) contains 1,bla,test,4 and seems to be an array internally
(which is not obvious when looking at the variable) so that $(internal:custom_testarray)[x] works.
But when I manipulate the content of $(internal:custom_testarray) directly, let's so to 2,bla,test,4 it seems to loose its
array status (which is again not obvious, it looks like before) and $(internal:custom_testarray)[x] does not work as expected
anymore.

Bottom line:
When looking at the variable content you can't tell if it is an array containing the 4 values 1,bla,test,4 or a string
"1,bla,test,4". If it is an array the array functionality breaks when manually changing its content.

**Comment by philipdroberts** [comment]
I was experiencing this same problem in v4.1.4 and was able to fix it by toggling the type of the variable in the "Custom Variables" screen to "{}" rather than "T"
